src/Zip.Shared/ZipFile.Check.cs

```diff
@@ -128,57 +128,59 @@ public static bool CheckZip(string zipFileName)
 
                 zip2 = ZipFile.Read(zipFileName);
 
-                foreach (var e1 in zip1)
+                foreach (var e2 in zip2)
                 {
-                    foreach (var e2 in zip2)
+                    var e1 = zip1._entries[e2.FileName];
```

Comment:
Won't we get ArgumentOutOfRangeException here if e2.FileName is not present in zip1._entries?

Reply:
Good catch! Actually it's KeyNotFoundException, at least that's what the docs say. It should be better now. Could you take another look?